lib/flappi/builder_factory.rb

```diff
@@ -76,12 +76,14 @@ def self.build_and_respond(controller)
       end
 
       # Merge in default values where one is defined and we don't have an actual parameter
-      controller.params.merge! Hash[controller.endpoint_info[:params].select do |defined_param|
-        !controller.params.key?(defined_param[:name]) && defined_param[:default]
-      end.
-      map do |defined_param|
-        [defined_param[:name], defined_param[:default]]
-      end]
+      controller.params.merge! Hash[
+        controller.endpoint_info[:params].select do |defined_param|
+          param = controller.params.dig(defined_param[:name])
```

Comment:
I've never seen an API with nested params using form syntax?

If we needed them, we could make up a DSL syntax and support them. I guess you'd be able to fudge a way of producing API docs. Does Ember ever generate nested params in queries?